src/components/ClickToView.tsx

```diff
@@ -0,0 +1,181 @@
+import { Button } from '@guardian/src-button';
+import { space } from '@guardian/src-foundations';
+import { background, border } from '@guardian/src-foundations/palette';
+import { headline, textSans } from '@guardian/src-foundations/typography';
+import { SvgCheckmark } from '@guardian/src-icons';
+import { css } from 'emotion';
+import React, { useState } from 'react';
+
+type RoleType =
+	| 'immersive'
+	| 'supporting'
+	| 'showcase'
+	| 'inline'
+	| 'thumbnail'
+	| 'halfWidth';
+
+export type ClickToViewProps = {
+	children: React.ReactNode;
+	role?: RoleType;
+	onAccept?: () => void;
+	source?: string;
+	sourceDomain?: string;
```

Comment:
Currently the only use of this callback in DCR is for the generic 'unsafe' embeds, the callback is used to kick off the iframe resizing logic once the click to view overlay has been clicked and the iframe has been actually rendered.

So I think this will probably be optional but I think that will fall out of the work that is going on currently.

Ill make the changes to use the Option convention.